README.md

```diff
@@ -123,81 +120,193 @@ $ pytest -v
-```
+These classes are based on the [`std::pmr::memory_resource`](https://en.cppreference.com/w/cpp/memory/memory_resource) interface class introduced in C++17 for polymorphic memory allocation.
 
-```
-// new
-rmmError_t result = RMM_ALLOC(&myvar, size_in_bytes, stream_id);
-// ...
-rmmError_t result = RMM_FREE(myvar, stream_id);
-```
+## `device_memory_resource`
 
-Note that `RMM_ALLOC` and `RMM_FREE` are wrappers around `rmm::alloc()` and
-`rmm::free()`, respectively. The lower-level functions also take a file name and
-a line number for tracking the location of RMM allocations and deallocations.
-The macro versions use the C preprocessor to automatically specify these params.
+`rmm::mr::device_memory_resource` is the base class that defines the interface for allocating and freeing device memory.
 
-### Using RMM with Thrust
+It has two key functions:
 
-RAPIDS and other CUDA libraries make heavy use of Thrust. Thrust uses CUDA device memory in two
-situations:
+1. `void* device_memory_resource::allocate(std::size_t bytes, cudaStream_t s)`
+   - Returns a pointer to an allocation of at least `bytes` bytes.
 
- 1. As the backing store for `thrust::device_vector`, and
- 2. As temporary storage inside some algorithms, such as `thrust::sort`.
+2. `void device_memory_resource::deallocate(void* p, std::size_t bytes, cudaStream_t s)`
+   - Reclaims a previous allocation of size `bytes` pointed to by `p`. 
+   - `p` *must* have been returned by a previous call to `allocate(bytes)`, otherwise behavior is undefined
+
+It is up to a derived class to provide implementations of these functions. 
+See [available resources](#availble-resources) for example `device_memory_resource` derived classes. 
+
+Unlike `std::pmr::memory_resource`, `rmm::mr::device_memory_resource` does not allow specifying an alignment argument. 
+All allocations are required to be aligned to at least 256B. 
+Furthermore, `device_memory_resource` adds an additional `cudaStream_t` argument to allow specifying the stream on which to perform the (de)allocation.
+
+### Available Resources
+
+RMM provides several `device_memory_resource` derived classes to satisfy various user requirements.
+For more detailed information about these resources, see their respective documentation.
+
+#### `cuda_memory_resource`
+
+Allocates and frees device memory using `cudaMalloc` and `cudaFree`.
+
+#### `managed_memory_resource`
+
+Allocates and frees device memory using `cudaMallocManaged` and `cudaFree`.
+
+#### `cnmem_(managed_)memory_resource`
 
-RMM includes a custom Thrust allocator in the file `thrust_rmm_allocator.h`. This defines the template class `rmm_allocator`, and
-a custom Thrust CUDA device execution policy called `rmm::exec_policy(stream)`.
+Uses the [CNMeM](https://github.com/NVIDIA/cnmem) pool sub-allocator to satisfy (de)allocations.
 
-#### Thrust Device Vectors
+#### `pool_memory_resource`
 
-Instead of creating device vectors like this:
+A coalescing, best-fit pool sub-allocator.
```

Comment:
Missing fixed-side resource and hybrid resource. in the interest of time, we could file an issue to document these later.